src/godot.cpp

```diff
@@ -271,7 +271,14 @@ GDExtensionBool GDExtensionBinding::init(GDExtensionInterfaceGetProcAddress p_ge
 	} else if (internal::godot_version.minor != GODOT_VERSION_MINOR) {
 		compatible = internal::godot_version.minor > GODOT_VERSION_MINOR;
 	} else {
+#if GODOT_VERSION_PATCH > 0
 		compatible = internal::godot_version.patch >= GODOT_VERSION_PATCH;
+#else
+		// Since internal::godot_version.patch is unsigned, it must be greater
+		// than or equal to GODOT_VERSION_PATCH which is zero. This avoids an
+		// error from -Werror=type-limits on GCC.
```

Comment:
Technically, it's a warning treated as error, not an error per se.
Personally I'd tend to be less verbose on comments for something that's not so important, e.g. `// Prevent -Wtype-limits warning due to unsigned comparison.`